contracts/gateway/EIP20Gateway.sol

```diff
@@ -477,7 +485,42 @@ contract EIP20Gateway is GatewayBase {
             bytes32(0),
             true
         );
+    }
+
+    /**
+     * @notice Return the revert penalty to the staker. Only valid for
+     *         a message transition from DeclaredRevocation -> Progressed.
+     *
+     * @dev Should only be called from progressStakeWithProof. This function
+     *      exists to avoid a stack too deep error.
+     *
+     * @param _messageHash Message hash.
+     * @param _staker Staker address.
+     * @param _oldMessageStatus Message status before progressing.
+     * @param _newMessageStatus Message status after progressing.
+     */
+    function tryReturnPenaltyToStaker(
+        bytes32 _messageHash,
+        address _staker,
+        MessageBus.MessageStatus _oldMessageStatus,
+        MessageBus.MessageStatus _newMessageStatus
+    )
+      private
+    {
+        if (_oldMessageStatus != MessageBus.MessageStatus.DeclaredRevocation) {
+            return;
+        }
+        if (_newMessageStatus != MessageBus.MessageStatus.Progressed) {
+            return;
+        }
+
+        // Penalty charged to staker for revert stake.
+        uint256 penalty = stakes[_messageHash].bounty
```

Comment:
We have a function to get penalty now.
`uint256 penalty = penaltyFromBounty(stakes[_messageHash].bounty);`